components/List/tests/unit/MDCSelfSizingStereoCellTests.m

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialList.h"
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialTypography.h"
+
+@interface MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow : UIWindow
+
+/** Used to override the value of @c preferredContentSizeCategory. */
+@property(nonatomic, copy) UIContentSizeCategory contentSizeCategoryOverride;
+
+@end
+
+@implementation MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = UIContentSizeCategoryLarge;
+  }
+  return self;
+}
+
+- (instancetype)initWithContentSizeCategoryOverride:
+    (UIContentSizeCategory)contentSizeCategoryOverride {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = contentSizeCategoryOverride;
+  }
+  return self;
+}
+
+- (UITraitCollection *)traitCollection {
+  if (@available(iOS 10.0, *)) {
+    UITraitCollection *traitCollection = [UITraitCollection
+        traitCollectionWithPreferredContentSizeCategory:self.contentSizeCategoryOverride];
+    return traitCollection;
+  }
+  return [super traitCollection];
+}
+
+@end
+
+@interface MDCSelfSizingStereoCellTests : XCTestCase
+
+@end
+
+@implementation MDCSelfSizingStereoCellTests
+
+- (void)testSelfSizingStereoCellDynamicTypeBehavior {
+  // Given
+  MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow *largeContainer =
+      [[MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow alloc]
+          initWithContentSizeCategoryOverride:UIContentSizeCategoryLarge];
+  MDCSelfSizingStereoCell *cell = [[MDCSelfSizingStereoCell alloc] init];
+  [largeContainer addSubview:cell];
+  UIFont *titleFont = cell.titleLabel.font;
+  UIFont *detailFont = cell.detailLabel.font;
+  MDCFontScaler *headlineFontScaler =
+      [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleSubtitle1];
+  // TODO: Investigate what can be done about the line below failing when using .caption instead of
+  // .button. See
+  // https://github.com/material-components/material-components-ios/pull/7413/files#r285289332 for
+  // context.
+  MDCFontScaler *detailFontScalar =
```

Comment:
Why do you need a custom font scaler? Can you not use the MDC Typography Scheme to provide scalable fonts for this test?

Reply:
I believe this is a test only for testing the cell dynamic type functionality. Scheme doesn't need to be introduced here for reusability.

Reply:
Yes, I know. It just seems like it'd be simpler to read the test if you didn't have to alloc/init a scaler, use a base font, get a scaled font, etc.  Instead just use a containerScheme with a DT TypographyScheme in setUp.

Reply:
There's a lot that can go wrong if every test is attempting to recreate the same conditions of a correctly-configured DT-compatible UIFont object.

Reply:
I am neutral on this one. But I feel introducing another layer of abstraction is not handy for debugging these tests if something changes in the scheme itself.